### ISSUE TITLE
Unable to compile due to recent updates?

### ISSUE THREAD

**Issue by holmes89** [WasmEdge/wasmedge-rust-sdk#60]
WasmEdge Version: 0.13.3
SDK version: 0.11.2
System: Ubuntu 20.04.5 LTS

I'm unsure what happened. I had my code compiling one day (08/25/2023) and without any changes (that I can remember) when compiling the next time (08/29/2023) I encountered some type errors that weren't there before:

```rust
error[E0425]: cannot find function `path_to_cstring` in module `utils`
  --> /home/jholmes/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmedge-sys-0.16.2/src/loader.rs:83:29
   |
83 |         let c_path = utils::path_to_cstring(file.as_ref())?;
   |                             ^^^^^^^^^^^^^^^ not found in `utils`

error[E0425]: cannot find function `path_to_cstring` in module `utils`
  --> /home/jholmes/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmedge-sys-0.16.2/src/plugin.rs:42:29
   |
42 |         let c_path = utils::path_to_cstring(path.as_ref())?;
   |                             ^^^^^^^^^^^^^^^ not found in `utils`

error[E0308]: mismatched types
   --> /home/jholmes/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmedge-sys-0.16.2/src/types.rs:210:54
    |
210 |             ctx: unsafe { ffi::WasmEdge_ValueGenV128(val) },
    |                           -------------------------- ^^^ expected struct `int128_t`, found `i128`
    |                           |
    |                           arguments to this function are incorrect
    |
note: function defined here
   --> /home/jholmes/personal/refactoring-rust/code/journal/target/wasm32-wasi/release/build/wasmedge-sys-aa676d8e52fbe445/out/wasmedge.rs:930:12
    |
930 |     pub fn WasmEdge_ValueGenV128(Val: int128_t) -> WasmEdge_Value;
    |            ^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /home/jholmes/.cargo/registry/src/github.com-1ecc6299db9ec823/wasmedge-sys-0.16.2/src/types.rs:217:18
    |
216 |     pub fn to_v128(&self) -> i128 {
    |                              ---- expected `i128` because of return type
217 |         unsafe { ffi::WasmEdge_ValueGetV128(self.ctx) }
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i128`, found struct `int128_t`

Some errors have detailed explanations: E0308, E0425.
For more information about an error, try `rustc --explain E0308`.
The following warnings were emitted during compilation:

warning: [wasmedge-sys] searching for existing libwasmedge install
warning: [wasmedge-sys] searching for libwasmedge at LibWasmEdgePaths { lib_dir: "/home/jholmes/.wasmedge/lib64", inc_dir: "/home/jholmes/.wasmedge/include" }
warning: [wasmedge-sys] searching for libwasmedge at LibWasmEdgePaths { lib_dir: "/home/jholmes/.wasmedge/lib", inc_dir: "/home/jholmes/.wasmedge/include" }
warning: [wasmedge-sys] found libwasmedge at LibWasmEdgePaths { lib_dir: "/home/jholmes/.wasmedge/lib", inc_dir: "/home/jholmes/.wasmedge/include" }
warning: [wasmedge-sys] generating bindgen header "/home/jholmes/personal/refactoring-rust/code/journal/target/wasm32-wasi/release/build/wasmedge-sys-aa676d8e52fbe445/out/wasmedge.rs"
```
My specific code looks like this:

```rust
fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut args: Vec<String> = env::args().skip(1).collect();
    let term = args.pop().unwrap_or("type".to_string());
    let res: Vec<String> = search(term, 0, 10).unwrap();
    for entry in res.iter() {
        println!("{:?}", entry);
    }    
    Ok(())
}

#[no_mangle]
pub fn static_search(
    page: isize,
    offset: isize,
) {
    
    let res: Vec<String> = search("type".to_string(), page, offset).unwrap();
    for entry in res.iter() {
        println!("{:?}", entry);
    }    
}
```
I've tried updating, rolling back, etc but nothing works. I'm unsure if my code was bad to begin with and something was cached or if something else happened so I'm unsure what I should do.

**Comment by holmes89** [comment]
I don't know if this is a red herring but I keep seeing this:

```github.com-1ecc6299db9ec823/wasmedge-sys-0.16.2```

In the stack which does not align (I believe) with my installed version. I've attempted to remove my Cargo.lock and clear the cache but this doesn't seem to work.

**Comment by holmes89** [comment]
So the problem was resolved when I changed the structure of my project. I'm trying to write up an example of a library, WASM binary, and consumer application all using Rust. When I had all three packages under one workspace there seemed to be an issue. When I moved the consuming application (with the SDK) into its own project the compilation works on the WASM binary.

I don't think anything more is needed. Thank you for your help!
